mlflow/fastai/callback.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import os
+import shutil
+import tempfile
+from functools import partial
+import matplotlib.pyplot as plt
+
+import mlflow.tracking
+from mlflow.utils.autologging_utils import try_mlflow_log
+from mlflow.fastai import log_model
+
+from fastai.callback.core import Callback
+
+
+# Move outside, because it cannot be pickled. Besides, ExceptionSafeClass was giving some issues
+class __MLflowFastaiCallback(Callback):
+    """
+    Callback for auto-logging metrics and parameters.
+    Records model structural information as params when training begins.
+    """
+
+    from fastai.learner import Recorder
+    from fastai.callback.all import TrackerCallback
+
+    remove_on_fetch, run_before, run_after = True, TrackerCallback, Recorder
+
+    def __init__(self, metrics_logger, log_models, is_fine_tune=False):
+        super().__init__()
+        self.metrics_logger = metrics_logger
+        self.log_models = log_models
+        self.is_fine_tune = is_fine_tune
+        self.freeze_prefix = ""
+
+    def after_epoch(self):
+        """Log loss and other metrics values after each epoch"""
+
+        def _is_float(x):
+            try:
+                float(x)
+                return True
+            except (ValueError, TypeError):
+                return False
+
+        # Do not record in case of predicting
+        if hasattr(self, "lr_finder") or hasattr(self, "gather_preds"):
+            return
+        # Remove non-float metrics and record the rest.
+        metrics = self.recorder.log
+        metrics = dict(zip(self.recorder.metric_names, metrics))
+        for key in [key for key, val in metrics.items() if not _is_float(val)]:
+            del metrics[key]
```

Comment:
```suggestion
        metrics = {k: v for k, v in zip(self.recorder.metric_names, metrics) if _is_float(v)}
```

Can we simplify this the code here?